sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs

```diff
@@ -180,6 +143,80 @@ private static ExportResult IsSuccess(HttpMessage httpMessage)
             return ExportResult.Failure;
         }
 
+        private static ApplicationInsightsRestClient InitializeRestClient(AzureMonitorExporterOptions options, ConnectionVars connectionVars, TokenCredential? credential)
+        {
+            HttpPipeline pipeline;
+
+            if (credential != null)
+            {
+                var scope = AadHelper.GetScope(connectionVars.AadAudience);
+                var httpPipelinePolicy = new HttpPipelinePolicy[]
+                {
+                    new BearerTokenAuthenticationPolicy(credential, scope),
+                    new IngestionRedirectPolicy()
+                };
+
+                pipeline = HttpPipelineBuilder.Build(options, httpPipelinePolicy);
+                AzureMonitorExporterEventSource.Log.WriteInformational("SetAADCredentialsToPipeline", $"HttpPipelineBuilder is built with AAD Credentials. TokenCredential: {credential.GetType().Name} Scope: {scope}");
+            }
+            else
+            {
+                var httpPipelinePolicy = new HttpPipelinePolicy[] { new IngestionRedirectPolicy() };
+                pipeline = HttpPipelineBuilder.Build(options, httpPipelinePolicy);
+            }
+
+            return new ApplicationInsightsRestClient(new ClientDiagnostics(options), pipeline, host: connectionVars.IngestionEndpoint);
+        }
+
+        private static PersistentBlobProvider? InitializeOfflineStorage(AzureMonitorExporterOptions options)
+        {
+            if (!options.DisableOfflineStorage)
+            {
+                try
+                {
+                    var storageDirectory = options.StorageDirectory
+                        ?? StorageHelper.GetDefaultStorageDirectory()
+                        ?? throw new InvalidOperationException("Unable to determine offline storage directory.");
+
+                    // TODO: Fallback to default location if location provided via options does not work.
+                    AzureMonitorExporterEventSource.Log.WriteInformational("InitializedPersistentStorage", storageDirectory);
+
+                    return new FileBlobProvider(storageDirectory);
+                }
+                catch (Exception ex)
+                {
+                    // TODO:
+                    // Remove this when we add an option to disable offline storage.
+                    // So if someone opts in for storage and we cannot initialize, we can throw.
+                    // Change needed on persistent storage side to throw if not able to create storage directory.
+                    AzureMonitorExporterEventSource.Log.WriteError("FailedToInitializePersistentStorage", ex);
+
+                    return null;
+                }
+            }
+
+            return null;
+        }
+
+        private static ConnectionVars InitializeConnectionVars(AzureMonitorExporterOptions options)
+        {
+            if (options.ConnectionString == null)
```

Comment:
Debatable topic :blush:

Which should take precedence - options set in the code or environment variable?

Reply:
The decision we made for AI SDK was anything hard-coded would take precedence over the env var.

Reply:
Adding @vishweshbankwar to this thread.
I'm not certain that there's anything to debate here. :)
I feel like this was settled when we built Connection Strings a few years ago.

If we need to debate this, I suggest we bring this up in the next Contributors Meeting and let all languages weigh in.

Reply:
question: how does OTel SDK handles it? I cannot recall. I think we just need to be consistent with it.

Reply:
> If we need to debate this, I suggest we bring this up in the next Contributors Meeting and let all languages weigh in. 👍

Reply:
Added this to the agenda for next week.